test/e2e/karmadactl_test.go

```diff
@@ -1311,6 +1311,203 @@ var _ = framework.SerialDescribe("Karmadactl taint testing", ginkgo.Labels{NeedC
 	})
 })
 
+var _ = framework.SerialDescribe("Karmadactl addons testing", ginkgo.Ordered, func() {
+	var karmadaHostContext = "karmada-host"
+	var kubeHostClient *kubernetes.Clientset
+	var member1 string
+
+	ginkgo.BeforeAll(func() {
+		member1 = framework.ClusterNames()[0]
+	})
+
+	ginkgo.BeforeAll(func() {
+		// Load kubeconfig file.
+		kubeconfigData, err := os.ReadFile(kubeconfig)
+		gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+		// Build a kubeconfig object specifying the "karmada-host" context.
+		config, err := clientcmd.Load(kubeconfigData)
+		gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+		// Set the context to "karmada-host"
+		config.CurrentContext = karmadaHostContext
+
+		// Build the REST client configuration for the "karmada-host" context.
+		restConfig, err := clientcmd.NewDefaultClientConfig(*config, &clientcmd.ConfigOverrides{}).ClientConfig()
+		gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+		// Initialize the kubeClient using the restConfig.
+		kubeHostClient, err = kubernetes.NewForConfig(restConfig)
+		gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+		// Retrieve the existing karmada-cert-secret.
+		karmadaCertSecret, err := kubeHostClient.CoreV1().Secrets(util.NamespaceKarmadaSystem).Get(context.TODO(), "karmada-cert-secret", metav1.GetOptions{})
+		gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+		// Create a new secret named karmada-cert with the same data.
+		karmadaCert := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      options.KarmadaCertsName,
+				Namespace: util.NamespaceKarmadaSystem,
+			},
+			Data: karmadaCertSecret.Data,
+			Type: karmadaCertSecret.Type,
+		}
+
+		_, err = kubeHostClient.CoreV1().Secrets(util.NamespaceKarmadaSystem).Create(context.TODO(), karmadaCert, metav1.CreateOptions{})
```

Comment:
@mohamedawnallah Are `karmada-cert` and `karmada-cert-secret` the same certificate used in two different installation methods? Since the names are different, you need to copy `karmada-cert-secret` and create `karmada-cert`.

Reply:
Yes, the certificates are used in the two different installation methods. I’m not sure if I’m missing something but I’ve already created the `karmada-cert` from the content of `karmada-cert-secret`? Here’s a snippet from the related [PR code](https://github.com/karmada-io/karmada/pull/5313/files#diff-9a6686056550152236595c63059c8df9e16e91721c9e30930ccacf236506645eR1344-R1361):

```go
...
// Retrieve the existing karmada-cert-secret.
karmadaCertSecret, err := kubeHostClient.CoreV1().Secrets(util.NamespaceKarmadaSystem).Get(context.TODO(), "karmada-cert-secret", metav1.GetOptions{})
gomega.Expect(err).ShouldNot(gomega.HaveOccurred())

// Create a new secret named karmada-cert with the same data.
karmadaCert := &corev1.Secret{
	ObjectMeta: metav1.ObjectMeta{
		Name:      options.KarmadaCertsName,
		Namespace: util.NamespaceKarmadaSystem,
	},
	Data: karmadaCertSecret.Data,
	Type: karmadaCertSecret.Type,
}

_, err = kubeHostClient.CoreV1().Secrets(util.NamespaceKarmadaSystem).Create(context.TODO(), karmadaCert, metav1.CreateOptions{})
...
```

Monitoring the creation and later cleanup of `karmada-cert` on `karmada-host` context:

```bash
$ watch -n 1 "kubectl get secrets -n karmada-system --context karmada-host"
NAME                  TYPE                DATA   AGE
karmada-cert          Opaque              14     2s
karmada-cert-secret   Opaque              14     24m
kubeconfig            Opaque              1      24m
member1-kubeconfig    Opaque              1      24m
member2-kubeconfig    Opaque              1      23m
member3-kubeconfig    Opaque              1      23m
webhook-cert          kubernetes.io/tls   2      24m
```

Reply:
Thank you for the detailed explanation; this indeed seems like a compromise. Issue #5363 is working on addressing this problem, and once it's resolved, we should be able to remove the process of duplicating the secret.